core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -82,7 +82,12 @@ public class PartitionsTable extends BaseMetadataTable {
                 10,
                 "last_updated_snapshot_id",
                 Types.LongType.get(),
-                "Id of snapshot that last updated this partition"));
+                "Id of snapshot that last updated this partition"),
+            Types.NestedField.required(
```

Comment:
And this topic always comes up, but what do you think of the position?  @ajantha-bhat @dramaticlly .  Maybe its better after file_count?  (so we have 3 columns for data, pos_delete, and eq_delete)

Reply:
Yeah I think what Szehon said make sense, given last 2 columns are optional and new are required

Reply:
Agree. I have already kept it beside `file_count` for partition stats. 

Note: Here we should not modify field id while reordering to maintain the compatibility.